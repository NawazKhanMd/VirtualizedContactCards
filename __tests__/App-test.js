/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import { ApplicationProvider, IconRegistry } from '@ui-kitten/components';

global.fetch = jest.fn(() => new Promise(resolve => resolve()));
jest.mock('react-native-gesture-handler', () => {});
jest.mock('@react-native-community/netinfo', () => {});


jest.useFakeTimers()

it('renders correctly', async  () => {
  const tree = renderer
  .create(<App />)
  .toJSON();
expect(tree).toMatchSnapshot();
});
