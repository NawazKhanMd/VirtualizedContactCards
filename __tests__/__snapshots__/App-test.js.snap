// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../images/bg.gif",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "transparent",
          "borderRadius": 15,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "space-between",
          "margin": 10,
          "padding": 10,
          "width": "95%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 15,
            "color": "black",
            "fontSize": 15,
            "height": 50,
            "letterSpacing": 1.2,
            "paddingLeft": 70,
            "textAlign": "center",
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <RCTScrollView
      data={
        Array [
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
          " ",
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={4}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.5}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#D47FA6",
                "borderRadius": 15,
                "display": "flex",
                "elevation": 10,
                "height": 444.6666666666667,
                "justifyContent": "flex-start",
                "marginBottom": -296.44444444444446,
                "marginHorizontal": 30,
                "marginTop": 15,
                "padding": 15,
                "transform": Array [
                  Object {
                    "rotateX": "-35deg",
                  },
                ],
                "width": 690,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 15,
                    "elevation": 3,
                    "height": 72,
                    "width": 72,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                  Object {
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 13,
                      "letterSpacing": 1.2,
                    },
                    Object {
                      "marginTop": 10,
                    },
                  ]
                }
              >
                See More
              </Text>
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={15}
                  bbWidth={15}
                  fill="#fff"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "height": 15,
                        "marginBottom": -12,
                        "width": 15,
                      },
                      Object {
                        "flex": 0,
                        "height": 15,
                        "width": 15,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M12 16a1 1 0 0 1-.64-.23l-6-5a1 1 0 1 1 1.28-1.54L12 13.71l5.36-4.32a1 1 0 0 1 1.41.15 1 1 0 0 1-.14 1.46l-6 4.83A1 1 0 0 1 12 16z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#8A56AC",
                "borderRadius": 15,
                "display": "flex",
                "elevation": 10,
                "height": 444.6666666666667,
                "justifyContent": "flex-start",
                "marginBottom": -296.44444444444446,
                "marginHorizontal": 30,
                "marginTop": 15,
                "padding": 15,
                "transform": Array [
                  Object {
                    "rotateX": "-35deg",
                  },
                ],
                "width": 690,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 15,
                    "elevation": 3,
                    "height": 72,
                    "width": 72,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                  Object {
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 13,
                      "letterSpacing": 1.2,
                    },
                    Object {
                      "marginTop": 10,
                    },
                  ]
                }
              >
                See More
              </Text>
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={15}
                  bbWidth={15}
                  fill="#fff"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "height": 15,
                        "marginBottom": -12,
                        "width": 15,
                      },
                      Object {
                        "flex": 0,
                        "height": 15,
                        "width": 15,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M12 16a1 1 0 0 1-.64-.23l-6-5a1 1 0 1 1 1.28-1.54L12 13.71l5.36-4.32a1 1 0 0 1 1.41.15 1 1 0 0 1-.14 1.46l-6 4.83A1 1 0 0 1 12 16z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#241332",
                "borderRadius": 15,
                "display": "flex",
                "elevation": 10,
                "height": 444.6666666666667,
                "justifyContent": "flex-start",
                "marginBottom": -296.44444444444446,
                "marginHorizontal": 30,
                "marginTop": 15,
                "padding": 15,
                "transform": Array [
                  Object {
                    "rotateX": "-35deg",
                  },
                ],
                "width": 690,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 15,
                    "elevation": 3,
                    "height": 72,
                    "width": 72,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                  Object {
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 13,
                      "letterSpacing": 1.2,
                    },
                    Object {
                      "marginTop": 10,
                    },
                  ]
                }
              >
                See More
              </Text>
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={15}
                  bbWidth={15}
                  fill="#fff"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "height": 15,
                        "marginBottom": -12,
                        "width": 15,
                      },
                      Object {
                        "flex": 0,
                        "height": 15,
                        "width": 15,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M12 16a1 1 0 0 1-.64-.23l-6-5a1 1 0 1 1 1.28-1.54L12 13.71l5.36-4.32a1 1 0 0 1 1.41.15 1 1 0 0 1-.14 1.46l-6 4.83A1 1 0 0 1 12 16z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#4aaaaa",
                "borderRadius": 15,
                "display": "flex",
                "elevation": 10,
                "height": 444.6666666666667,
                "justifyContent": "flex-start",
                "marginBottom": -296.44444444444446,
                "marginHorizontal": 30,
                "marginTop": 15,
                "padding": 15,
                "transform": Array [
                  Object {
                    "rotateX": "-35deg",
                  },
                ],
                "width": 690,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 15,
                    "elevation": 3,
                    "height": 72,
                    "width": 72,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                  Object {
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 13,
                      "letterSpacing": 1.2,
                    },
                    Object {
                      "marginTop": 10,
                    },
                  ]
                }
              >
                See More
              </Text>
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={15}
                  bbWidth={15}
                  fill="#fff"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "height": 15,
                        "marginBottom": -12,
                        "width": 15,
                      },
                      Object {
                        "flex": 0,
                        "height": 15,
                        "width": 15,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M12 16a1 1 0 0 1-.64-.23l-6-5a1 1 0 1 1 1.28-1.54L12 13.71l5.36-4.32a1 1 0 0 1 1.41.15 1 1 0 0 1-.14 1.46l-6 4.83A1 1 0 0 1 12 16z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
